eth/downloader/fetchers.go

```diff
@@ -45,9 +45,6 @@ func (d *Downloader) fetchHeadersByHash(p *peerConnection, hash common.Hash, amo
 	defer timeoutTimer.Stop()
 
 	select {
-	case <-d.cancelCh:
```

Comment:
This is necessary, otherwise this API will be useless if downloader finishes a sync cycle (cancel channel will be closed)